pyanaconda/payload/dnfpayload.py

```diff
@@ -912,6 +912,9 @@ def environmentDescription(self, environmentid):
 
     def environmentId(self, environment):
         """Return environment id for the environment specified by id or name."""
+        # the enviroment must be string or else DNF throws an assert error
+        if not isinstance(environment, str):
+            raise TypeError
```

Comment:
If `None` is an acceptable value for anaconda, then just test if `environment is None` and return `None`. It is dangerous to silently ignore all types of the environment.

Reply:
And also log this please.

Reply:
Makes sense, I've updated to PR to:
- shortcut on None to return None
- logged a warning if someone does try to feed the payload an environment that is not a string